src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1206,29 +1280,44 @@ static ucs_status_t uct_ib_mlx5dv_check_dc(uct_ib_device_t *dev)
     attr.ah_attr.grh.traffic_class = 0;
     attr.ah_attr.grh.sgid_index    = 0;
 
-    ret = ibv_modify_qp(qp, &attr, IBV_QP_STATE |
-                                   IBV_QP_MIN_RNR_TIMER |
-                                   IBV_QP_AV |
-                                   IBV_QP_PATH_MTU);
+    ret = ibv_modify_qp(dct_qp, &attr, IBV_QP_STATE |
+                                       IBV_QP_MIN_RNR_TIMER |
+                                       IBV_QP_AV |
+                                       IBV_QP_PATH_MTU);
 
     if (ret != 0) {
         ucs_debug("%s: failed to ibv_modify_qp(DCT, RTR): %m",
                   uct_ib_device_name(dev));
         status = UCS_OK;
-        goto out_destroy_qp;
+        goto out_destroy_dct_qp;
+    }
+
+    uct_ib_mlx5dv_qp_init_attr(&qp_init_attr, srq, pd, cq, IBV_QPT_DRIVER);
+    uct_ib_mlx5dv_dc_qp_init_attr(&dv_attr, MLX5DV_DCTYPE_DCI);
+    dci_qp = UCS_PROFILE_CALL_ALWAYS(mlx5dv_create_qp, ctx, &qp_init_attr,
+                                     &dv_attr);
+    if (dci_qp == NULL) {
+        ucs_debug("%s: mlx5dv_create_qp(DCT) failed: %m",
+                  uct_ib_device_name(dev));
+        status = UCS_OK;
```

Comment:
is this really ok? Here we already assume that DC is supported, but we can't determine the queues length

Reply:
ok